packages/turf-meta/index.js

```diff
@@ -822,6 +822,128 @@ function feature(geometry, properties) {
     };
 }
 
+
+/**
+ * Callback for segmentEach
+ *
+ * @callback segmentEachCallback
+ * @param {Feature<LineString>} [currentSegment] The current segment being processed.
+ * @param {number} [currentIndex] The index of the current element being processed in the array, starts at index 0.
+ * @returns {void}
+ */
+
+/**
+ * Iterate over 2-vertex line segment in any GeoJSON object, similar to Array.forEach()
+ *
+ * @param {FeatureCollection|Feature|Geometry} geojson any GeoJSON
+ * @param {Function} callback a method that takes (currentSegment, currentIndex)
+ * @returns {void}
+ * @example
+ * var polygon = {
+ *   "type": "Feature",
+ *   "properties": {},
+ *   "geometry": {
+ *     "type": "Polygon",
+ *     "coordinates": [[[-50, 5], [-40, -10], [-50, -10], [-40, 5], [-50, 5]]]
+ *   }
+ * }
+ * // Iterate over GeoJSON by 2-vertex segments
+ * turf.segmentEach(polygon, function (currentSegment, currentIndex) {
+ *   //= currentSegment
+ *   //= currentIndex
+ * });
+ *
+ * // Calculate the total number of segments
+ * var total = 0;
+ * var initialValue = 0;
+ * turf.segmentEach(polygon, function () {
+ *     total++;
+ * }, initialValue);
+ */
+function segmentEach(geojson, callback) {
+    var currentIndex = 0;
+    flattenEach(geojson, function (feature) {
+        coordReduce(feature, function (previousCoords, currentCoords) {
+            var currentSegment = {
```

Comment:
@DenisCarriere I bet it's a choice not to have dependencies in `turf-meta`.
Otherwise you could use
```js
var currentSegment = lineString([previousCoords, currentCoords], feature.properties || {});
```